buildbot/configure.py

```diff
@@ -11,30 +11,49 @@ def do_configure(args):
     sycl_dir = os.path.join(args.src_dir, "sycl")
     spirv_dir = os.path.join(args.src_dir, "llvm-spirv")
     ocl_header_dir = os.path.join(args.obj_dir, "OpenCL-Headers")
-    icd_loader_lib = ''
+    icd_loader_lib = os.path.join(args.obj_dir, "OpenCL-ICD-Loader", "build")
+    llvm_targets_to_build = 'X86'
+    llvm_enable_projects = 'clang;llvm-spirv;sycl'
+    libclc_targets_to_build = ''
+    sycl_build_pi_cuda = 'OFF'
+    llvm_enable_assertions = 'OFF'
 
     if platform.system() == 'Linux':
-      icd_loader_lib = os.path.join(args.obj_dir, "OpenCL-ICD-Loader", "build", "libOpenCL.so")
+        icd_loader_lib = os.path.join(icd_loader_lib, "libOpenCL.so")
     else:
-      icd_loader_lib = os.path.join(args.obj_dir, "OpenCL-ICD-Loader", "build", "OpenCL.lib")
+        icd_loader_lib = os.path.join(icd_loader_lib, "OpenCL.lib")
+
+    if args.cuda:
+        llvm_targets_to_build += ';NVPTX'
+        llvm_enable_projects += ';libclc'
+        libclc_targets_to_build = 'nvptx64--;nvptx64--nvidiacl'
+        sycl_build_pi_cuda = 'ON'
+
+    if args.assertions:
+        llvm_enable_assertions = 'ON'
 
     install_dir = os.path.join(args.obj_dir, "install")
 
-    cmake_cmd = ["cmake",
-                 "-G", "Ninja",
-                 "-DCMAKE_BUILD_TYPE={}".format(args.build_type),
-                 "-DLLVM_EXTERNAL_PROJECTS=sycl;llvm-spirv;opencl-aot",
-                 "-DLLVM_EXTERNAL_SYCL_SOURCE_DIR={}".format(sycl_dir),
-                 "-DLLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR={}".format(spirv_dir),
-                 "-DLLVM_ENABLE_PROJECTS=clang;sycl;llvm-spirv;opencl-aot",
-                 "-DOpenCL_INCLUDE_DIR={}".format(ocl_header_dir),
-                 "-DOpenCL_LIBRARY={}".format(icd_loader_lib),
-                 "-DLLVM_BUILD_TOOLS=ON",
-                 "-DSYCL_ENABLE_WERROR=ON",
-                 "-DLLVM_ENABLE_ASSERTIONS=ON",
-                 "-DCMAKE_INSTALL_PREFIX={}".format(install_dir),
-                 "-DSYCL_INCLUDE_TESTS=ON", # Explicitly include all kinds of SYCL tests.
-                 llvm_dir]
+    cmake_cmd = [
+        "cmake",
+        "-G", "Ninja",
+        "-DCMAKE_BUILD_TYPE={}".format(args.build_type),
+        "-DLLVM_ENABLE_ASSERTIONS={}".format(llvm_enable_assertions),
+        "-DLLVM_TARGETS_TO_BUILD={}".format(llvm_targets_to_build),
+        "-DLLVM_EXTERNAL_PROJECTS=sycl;llvm-spirv",
```

Comment:
opencl-aot project is missed

Reply:
Is this needed for the overall project as well or just for the testing? Its not mentioned on the Getting Started Guide

Reply:
@Ruyk opencl-aot is an optional tool to enable AOT compilation for SYCL. It is not a requirement for end users, but it is being tested in CI.